src/teleport.cpp

```diff
@@ -70,13 +70,36 @@ void Teleport::addThing(Thing* thing)
 	return addThing(0, thing);
 }
 
+bool Teleport::checkInfinityLoop(Tile* destTile)
+{
+	if (!destTile) {
```

Comment:
this check is redudant, cause you check that in the method which you call that one.